src/thunderbots/software/ai/navigator/obstacle/obstacle.cpp

```diff
@@ -0,0 +1,109 @@
+
+#include "obstacle.h"
+
+const Polygon& Obstacle::getBoundaryPolygon() const
+{
+    return _polygon;
+}
+
+Obstacle::Obstacle() : _polygon{Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 0)} {}
+
+Obstacle::Obstacle(Polygon polygon) : _polygon(polygon) {}
+
+Obstacle Obstacle::createPlaceholderObstacle()
+{
+    return Obstacle();
+}
+
+Obstacle Obstacle::createRobotObstacle(const Robot& robot, bool enable_velocity_cushion)
+{
+    // TODO: figure out how to get tick length
```

Comment:
One way to get tick length is including <time.h> and doing 1/CLOCKS_PER_SEC

Reply:
I believe this is referring to an "AI tick" not the system clock.

Reply:
This tick length is to project how far the robots can travel between the current call of navigator and the next call of navigator. I decided that obstacle shouldn't figure out what the AI tick is, so making it a parameter to pass in from navigator

Reply:
Ah OK, disregard my comment then